### PR TITLE
[WIP] Mercadopago gateway implementation

### LINKED ISSUE #729: Implement Mercado Pago for Latam
The lib [react-native-mercadopago-px](https://github.com/BlackBoxVision/react-native-mercadopago-px) seems pretty good, and actively maintained. 

On server-side, it would need to implement the "preferences" API (as described in the docs)

Also, watch BlackBoxVision/react-native-mercadopago-px#14 to implement Marketplace flow.

### PR DESCRIPTION
**Pull request by sgobotta**
### This is a work in progress, do not merge.

Depends on [coopcycle-web/pull/1655](https://github.com/coopcycle/coopcycle-web/pull/1655)

Closes #729 

**Suggestion to repository maintainers:**

+ create a `mercadopago` branch from master
+ edit the target branch of this PR
+ delete this message afterwards

**This PR provides:**

+ [x] Basic checkout using [react-native-mercadopago-px](https://github.com/BlackBoxVision/react-native-mercadopago-px) using a new component `MercadopagoCheckout`
+ [x] `MoreInfo` component refactor to choose between gateways

**Pending tasks:**

+ [x] Refactor the `getPreferenceId` function to an http call to the server. The preferences Mercadopago request should be done server-side
+ [x] Process the returned order from the server after the `mercadopagoCheckout` finishes
+ [ ] Modify `react-native-mercadopago-px` usage to support ~split payments~ the auth/capture payment flow. *This PR won't include such functionality*

### CODE REVIEW

File: android/build.gradle [hunk 1]
```diff
@@ -3,7 +3,7 @@
 buildscript {
     ext {
         buildToolsVersion = "28.0.3"
-        minSdkVersion = 18
+        minSdkVersion = 19
```
Comment (alexsegura) [inline android/build.gradle:6]: Why is this needed? Some people have quite old devices, see #745

Comment (alexsegura) [reply android/build.gradle:6]: Ok sorry, I didn't see the comment of 2373df7
I see, it appeared in [version 4.29.0](https://github.com/mercadopago/px-android/releases/tag/4.29.0)

Comment (alexsegura) [reply android/build.gradle:6]: Another possibility would be to override the version, like:

```
mercadoPagoPxVersion = "4.28.0"
```

But it doesn't seem like a good idea... @jamesdgrant are there lots of messengers with Android 4.3?

**Comment by sgobotta** [comment]
@alexsegura I'm fixing conflicts with master and encountered changes in the android build tools version. Are you probably using a newer gradle version? Here's some of the output:

```bash
> Task :app:processDebugManifest FAILED

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.0.1/userguide/command_line_interface.html#sec:command_line_warnings
711 actionable tasks: 5 executed, 706 up-to-date

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:processDebugManifest'.
> java.lang.NullPointerException (no error message)
```

Current build tools versions:

```
buildToolsVersion = "29.0.2"
minSdkVersion = 19
compileSdkVersion = 29
targetSdkVersion = 29
```

Current gradle version:

```
Gradle 6.0.1
```

Run the restart script in `README.md` with no success. I'll be trying other versions, please let me know if you used any other specific configuration in the last master build

**Comment by sgobotta** [comment]
This PR depends on https://github.com/facttic/coopcycle-web/commit/84d2593832bbbc85fd65faca2234d88854becc17
